### ISSUE TITLE
Table event_publication: value too long for type

### ISSUE THREAD

**Issue by ivanfryda** [spring-projects/spring-modulith#471]
When saving an Event in PostgreSQL I get the following error:

Caused by: org.hibernate.exception.DataException: could not execute statement [ERROR: value too long for type character varying(255)] [insert into event_publication (completion_date,event_type,listener_id,publication_date,serialized_event,id) values (?,?,?,?,?,?)]

So then I forced that column to be of type bytea by creating an orm.xml in META-INF. Then I got this error:

org.hibernate.exception.SQLGrammarException: could not execute statement [ERROR: column "serialized_event" is of type bytea but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 129] [insert into event_publication (completion_date,event_type,listener_id,publication_date,serialized_event,id) values (?,?,?,?,?,?)]

How to solve this?

**Comment by odrotbohm** [comment]
How do you create the schema for the event publications? We generally recommend to not let a JPA provider create the schema. The JDBC module recommends [this](https://docs.spring.io/spring-modulith/reference/appendix.html#schemas.postgresql) for PostgreSQL.

**Comment by ivanfryda** [comment]
Exactly, I am letting the JPA provider create the schema. I now fixed it by having it compliant with the documentation as provided with this orm.xml:

```
<entity-mappings xmlns="http://xmlns.jcp.org/xml/ns/persistence/orm"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                 xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence/orm
                                     http://xmlns.jcp.org/xml/ns/persistence/orm_2_2.xsd"
                 version="2.2">
    <package>org.springframework.modulith.events.jpa</package>
    <entity class="JpaEventPublication">
        <attributes>
            <basic name="listenerId">
                <column name="listener_id" column-definition="text"/>
            </basic>
            <basic name="eventType">
                <column name="event_type" column-definition="text"/>
            </basic>
            <basic name="serializedEvent">
                <column name="serialized_event" column-definition="text"/>
            </basic>
        </attributes>
    </entity>
</entity-mappings>
```

**Comment by brunpoern** [comment]
I run into the same issue with MySQL. I got how to fix it (either by overwriting definition in orm.xml or create event_publication table upfront according to documentation). 

Just wondering why the `id` field in class `JpaEventPublication`  is defined with ` @Column(length = 16)`. Doesn't this mean that it fails by definition because the generated UUID never fits into a column with max lenght 16?! Shouldn't length be 36 here to be consistent with database schema defined in documentation? Or do I miss something? I'm not a JPA expert to be honest.

**Comment by kromit** [comment]
I've run into this problem also usung JPA. In my case I am only trying to initialise a H2 database for testing. I've also tried H2 postgresql mode with postgres dialect with same result. The column `SERIALIZED_EVENT` is always created as `VARYING(255)` which is very short. I guess this is highly dependent on used hibernate dialect, but i think there should be at least `@Column(length = 4000)` defined for `serializedEvent` field in `JpaEventPublication`, since this same length is also documented for JDBC table definitions in SQL.
